packages/web/src/js/Collapse.ts

```diff
@@ -0,0 +1,167 @@
+}
+interface CollapseState {
+  collapsed: boolean;
+  width: number;
+  triggered: boolean;
+}
+
+class Collapse extends BaseComponent {
+  target: HTMLElement | null | undefined;
+  parent: HTMLElement | null | undefined;
+  meta: CollapseMeta;
+  state: CollapseState;
+
+  constructor(element: HTMLElement) {
+    super(element);
+    this.target = this.element.dataset.target ? selectorEngine.findOne(`#${this.element.dataset.target}`) : null;
+    this.parent = this.element && this.element.parentElement;
+    this.meta = {
+      id: this.element.dataset.target,
+      hideOnCollapse: !!(this.element.dataset.more || this.element.dataset.more === ''),
+      breakpoint: (this.target?.dataset?.breakpoint as breakpointType) || breakpointKeys.default,
+    };
+    this.state = {
+      collapsed: !!(
+        this.element.getAttribute(NAME_ARIA_EXPANDED) || this.element.classList.contains(CLASSNAME_EXPANDED)
+      ),
+      width: window.innerWidth,
+      triggered: false,
+    };
+
+    this.onInit();
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  onResize() {
+    this.state.width = window.innerWidth;
+    this.updateTriggerElementHandler();
+    this.updateCollapsibleElementHandler();
+  }
+
+  appendNodeToParentHandler() {
+    if (this.state.collapsed) {
+      const contentEl = this.target?.children[0];
+      const innerHtml = contentEl ? contentEl?.innerHTML : this.target?.innerHTML;
+      const originalHtml = this.parent?.innerHTML;
+      if (this.parent) {
+        this.parent.innerHTML = `${originalHtml}${innerHtml}`;
+      }
+      setTimeout(() => this.target?.remove(), 100);
+    }
+  }
+
+  adjustCollapsibleElementHeightHandler(collapsed: boolean = this.state.collapsed) {
+    if (this.target) {
+      const content = this.target.children[0];
+      const bounds = content.getBoundingClientRect();
+      this.target.style.height = collapsed ? `${bounds.height}px` : '0';
+    }
+  }
+
+  updateTriggerElementHandler(collapsed: boolean = this.state.collapsed) {
+    const triggers = selectorEngine.findAll(`[${NAME_DATA_TARGET}="${this.meta.id}"]`);
+    const updateElement = (element: Element | HTMLElement) => {
+      element.setAttribute(NAME_ARIA_CONTROLS, this.meta.id);
+      element.setAttribute(NAME_ARIA_EXPANDED, String(collapsed));
+      element.classList.toggle(CLASSNAME_EXPANDED, collapsed);
+      if (this.meta.hideOnCollapse && collapsed) {
+        element.remove();
+        this.appendNodeToParentHandler();
+        this.onDestroy();
+      }
+    };
+    this.state.collapsed = collapsed;
+    if (triggers.length === 1) {
+      updateElement(this.element);
+    } else {
+      triggers.forEach((trigger) => updateElement(trigger));
+    }
+  }
+
+  updateCollapsibleElementHandler(collapsed: boolean = this.state.collapsed) {
+    if (this.target) {
+      this.target.classList.toggle(CLASSNAME_COLLAPSED, collapsed);
+      this.adjustCollapsibleElementHeightHandler(collapsed);
+    }
+  }
+
+  setShow() {
```

Comment:
Please try to respect implementation in other components. We want to be consistent in writing the code. So please use `show()`, `hide()` and `toggle()` methods.